File: CPP_Programming/Day_4/program21.cpp
//Description : Write a program tha accepts number from the user & count the 
//              frequency of the digit

#include<iostream>
using namespace std;

class Digits
{
    public :
        int iNo;
        int iTarget;

    Digits(int iNo , int iTarget)
    {
        this->iNo = iNo;
        this->iTarget = iTarget;
    }

    int CountFrequency()
    {
        int iDigit = 0;
        int iCount = 0;

        while(iNo != 0)
        {
            iDigit = iNo % 10;
            if(iDigit == iTarget)
            {
                iCount++;
            }
            iNo = iNo /10;
        }

        return iCount;
    }
};


int main()
{
    int iValue = 0;
    int iRet = 0;
    int iTarget = 0;

    cout<<"Enter the number : ";
    cin>>iValue;

    cout<<"Enter the target : ";
    cin>>iTarget;

    Digits dobj(iValue , iTarget);

    iRet = dobj.CountFrequency();

    cout<<"Frequency is  : "<<iRet<<"\n";

    return 0;
}

File: Java_Programming/Day_4/program21.java
//Description : Write a program tha accepts number from the user & count the 
//              frequency of the digit

import java.util.*;
class Digits
{
    public int iNo;
    public int iTarget;

    Digits(int iNo , int iTarget)
    {
        this.iNo = iNo;
        this.iTarget = iTarget;
    }

    public int Frequency()
    {
        int iDigit = 0;
        int iCount = 0;

       while(iNo != 0)
       {
            iDigit = iNo % 10;
            if(iDigit == iTarget)
            {
                iCount++;
            }
            iNo = iNo / 10;
       }

       return iCount;
    }

}

class program21
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the number : ");
        int iValue = sobj.nextInt();

        System.out.println("Enter the target : ");
        int iTarget = sobj.nextInt();


        Digits dobj = new Digits(iValue , iTarget);

        int iRet = dobj.Frequency();

        System.out.println("Frequency is : "+iRet);

        sobj.close();
    }
}